app/controllers/orders_controller.rb

```diff
@@ -1,24 +1,70 @@
 class OrdersController < ApplicationController
   def cart # One cart (@order) holds information on zero or many OrderItems
-    @order = Order.new(status: "pending")
-    @items = OrderItem.all
+    @order = Order.find_by(status: "pending") # under the assumption there is only one order happening at a time, does this logic work?
+    if @order && @order.order_items.length != 0
+      @items = OrderItem.where(order_id: @order.id)
+    end
   end
 
   def new_order_item
     # TODO: Quantity should come from product#show view; hardcoded for now
-    @item = OrderItem.new(quantity: 1, order_id: @order.id, product_id: params[:id])
+    if Order.find_by(status: "pending")
+      @order = Order.find_by(status: "pending")
+    else
+      @order = Order.create(status: "pending")
+    end
+
+    @item = OrderItem.create(quantity: 1, order_id: @order.id, product_id: params[:id])
+    @order.order_items << @item
+    flash[:success] = "#{Product.find_by(id: @item.product_id).name} added to the shopping cart."
+    redirect_to product_path(params[:id])
   end
 
   def new #this should gather info for order's name, email, address, cc, etc...
+    @order = Order.find_by(status: "pending")
   end
 
   def update #this should update order with info above
+    @order = Order.find_by(status: "pending")
+    @order.status = "paid"
+    @order.save
+
+    is_successful = @order.update(order_params)
+    if is_successful
+      flash[:success] = "TESTTEST: order has been updated."
+    else
+      @order.errors.messages.each do |field, messages|
+        flash.now[field] = messages
+      end
+      render :new, status: :bad_request
+    end
+
+    redirect_to order_path(params[:id])
+  end
```

Comment:
curious about what this method is doing (looks good just looking for clarification). update occurs once the user puts in their payment info?

Reply:
My logic here: once user fills in checkout info and submits, our program should hit orders#update then orders#show so the user sees the confirmation page but the backend program is updating the order instance with update info (name, address, email, etc.) and sending everything to orders#show for confirmation and status change. There may be holes in this logic so please let me know if you see any!!